package.json

```
@@ -32,6 +32,6 @@
     "typedoc": "^0.25.3"
   },
   "dependencies": {
-    "typescript": "~5.3.3"
+    "typescript": "^5.4.2"
```

```suggestion
    "typescript": "~5.4.2"
```

What do you think about avoiding the caret operator here so that future regressions aren't implicitly propagated to existing releases? Using the caret would cause regressions from upstream TypeScript version updates between [5.5, 6.0) to automatically affect our releases.

Because TypeScript major version updates correspond to "minor" versions in semver, using the tilde operator here would allow us to test new TypeScript versions for regressions before upgrading them in our dependencies list.

---

This rationale is sensible to me; good looking out.